node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -551,22 +549,36 @@ func (atp *apiTransactionProcessor) getVirtualNonce(address []byte, accountsAdap
 		TransactionsProcessor: &txProcessor,
 	}
 
-	// TODO brainstorm if the selection session is needed
+	// the SelectionSession is used in this flow for fallbacks (e.g. the account does not exist in the proposed blocks, unexpected errors etc.)
```

Comment:
Can we maybe rely solely on the accounts adapter API in this case, without creating / passing the selection session? Maybe the following interface (from package `common`) can help: `AccountNonceProvider`.

Reply:
Now I realize - maybe we should do this first, since it's simpler (CC: @sstanculeanu): le't return an error if the account does not exist in the proposed blocks, for now. No virtual nonce. For non-virtual nonces, there's an existing endpoint. If we decide we need a unified function (endpoint), we'll do a distinct one, which calls first for virtual nonces, then for non-virtual nonces (fallback).